deliverylocation/models.py

```diff
@@ -0,0 +1,146 @@
+                                validators=[MinLengthValidator(6)],
+                                null=False,
+                                help_text="Password")
+    user_type = models.CharField(max_length=16,
+                                 validators=[MinLengthValidator(6)],
+                                 null=False,
+                                 choices=[('client', 'client'), ('supplier', 'supplier')],
+                                 help_text="User Type - supplier/client")
+
+    class Meta:
+        abstract = True
+
+
+class Supplier(User):
+    """Supplier class inherit from User abstract class.
+
+    Supplier class lets the app user to do variety of actions
+    which evolves the Product and Delivery class.
+
+    Attributes:
+        business_name: a string which holds the Supplier business name.
+    """
+    business_name = models.CharField(max_length=32,
+                                     validators=[MinLengthValidator(6)],
+                                     null=False,
+                                     help_text="Business Name")
+
+    @staticmethod
+    def filter_by_field(field, field_value):
+        """Fetches list of Suppliers filtered with given field and given value.
+
+        Args:
+        field:
+            Supplier attribute.
+        field_value:
+            value which will filter the suppliers.
+
+        Returns:
+            returns a queryset containing the objects that match the specified filter conditions.
+
+        Raises:
+        FieldDoesNotExist error: if Supplier doesn't have the specified attribute.
+        """
+        if hasattr(Supplier, str(field)):
+            return Supplier.objects.filter(**{field: field_value})
+        raise FieldDoesNotExist
+
+    @staticmethod
+    def get_all_suppliers():
+        """Fetches list of all Suppliers.
+
+        Args:
+            None.
+
+        Returns:
+            returns a queryset containing all the Suppliers.
+        """
+        return Supplier.objects.all()
+
+    @staticmethod
+    def get_all_fields():
+        """Fetches list of all Supplier fields.
+
+        Args:
+            None.
+
+        Returns:
+            returns a list of all Supplier fields.
+        """
+        return ['user_name', 'first_name', 'last_name',
+                'password', 'user_type', 'business_name']
+
+
+class DeliveryLocation(models.Model):
+    user_name = models.ForeignKey(Supplier, on_delete=models.CASCADE)
+    location = models.CharField(max_length=32, validators=[MinLengthValidator(1)])
+    date = models.DateField()
+
+    def add_delivery_location(self):
+        self.save()
+        try:
+            DeliveryLocation.full_clean(self)
+        except ValidationError as e:
+            self.delete()
+            raise e
+        return self
+
+    def remove_delivery_location(self):
+        DeliveryLocation.objects.get(id=self.id).delete()
+
+    def update_delivery_location_date(self, newdate):
```

Comment:
```suggestion
    def update_date(self, newdate):
```
As this is a method of the `DeliveryLocation` object, then I think you can drop the `delivery_location` from the name. It will make things look cleaner.

Reply:
Done